---

title: "hibernate CacheException interface changed"
layout: issue
tags: 
permalink: /browse/EHC-204

issue_key: EHC-204
issue_numeric_sort_key: 204
issuetype: "Bug"
project: "Ehcache Core"
project_key: "EHC"
status: "Closed"
resolution: "Fixed"
priority: ""
components: ""
labels: 
assignee: "drb"
reporter: "sourceforgetracker"
votes:  0
watchers: 0

created: "2009-09-21T15:07:53.000-0400"
updated: "2009-09-22T23:44:37.000-0400"
resolved: "2009-09-22T23:44:37.000-0400"




---

{% raw %}

## Description

<div markdown="1" class="description">

The interface to hibernate CacheException has changed. Currently ehcache is compiled against hibernate 3.2.1 when the constructor CacheException(Exception) existed. Now the signature for the constructor is CacheException(Throwable). This means that if you use the current version of hibernate then ehcache will always throw a MethodNotFound exception everytime it encounters a real exception because it will try to call the constructor CacheException(Exception). As a result we will never see the true exception ehcache is throwing. The only resolution is for every user to recompile ehcache.

ehcache needs to start compiling against a later version of hibernate to resolve this.

Sourceforge Ticket ID: 2035287 - Opened By: nobody - 1 Aug 2008 21:02 UTC

</div>

## Comments


{:.comment-heading}
### **Sourceforge Tracker** <span class="date">2009-09-21</span>

<div markdown="1" class="comment">

Ok, upgraded provided scope dependency to 3.2.6ga. The expected signature is Throwable.

Fixed in trunk and will be released in ehcache-1.6beta4
Comment by: gregluck - 4 Apr 2009 07:19 UTC

</div>


{:.comment-heading}
### **Fiona OShea** <span class="date">2009-09-22</span>

<div markdown="1" class="comment">

Re-opening so that I can properly close out these issues and have correct Resolution status in Jira

</div>



{% endraw %}
